UOP1_Project/Assets/Scripts/Characters/Character.cs

```diff
@@ -87,6 +88,24 @@ private void Update()
         movementVector.y = verticalMovement;
         characterController.Move(movementVector * Time.deltaTime);
 
+        if (characterController.isGrounded) {
+            //Check if underneath the character is stable
+            isUnderneathStable = false; //By default, underneath is not stable
+            RaycastHit hit;
+            Ray ray = new Ray(transform.position + characterController.center, Vector3.down);
+            if (Physics.Raycast(ray, out hit, Mathf.Infinity)) {
+                float underneathSlopeAngle = Vector3.Angle(transform.up, hit.normal);
+                if (underneathSlopeAngle <= characterController.slopeLimit) {
+                    //We found a stable underneath
+                    isUnderneathStable = true;
+                } else {
+                    //Add a slide movement along the underneath slope direction
+                    Vector3 underneathSlopeDirection = Vector3.Cross(hit.normal, Vector3.Cross(hit.normal, Vector3.up));
+                    characterController.Move(underneathSlopeDirection * gravityMultiplier * Time.deltaTime);
```

Comment:
Would be better to have just one call to `characterController.Move()`.
You could add `underneathSlopeDirection * gravityMultiplier` on `movementVector` before line 89.

Reply:
I decided to not add in the `movementVector` because it will affect the rotation of the character when the character slides down. I felt it was more natural (looking) to slide down by keeping the same rotation.
At the end, this should depend on the game design.